### PR TITLE
Add Kotlin Lambda (CDK) example

### PR DESCRIPTION
**Pull request by pcholakov**
This PR:

- adds a new Kotlin example deploying the hello-world Lambda handler with CDK
- assumes customer has a managed Restate cluster provided by us
- I kept this under Kotlin for now; I think we should do this for other CDK examples too. Some day we'll enable polyglot CDK constructs and we can redefine the app/stack in Kotlin

### CODE REVIEW

File: kotlin/hello-world-lambda-cdk/lambda/src/main/resources/log4j2.properties [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Set to debug or trace if log4j initialization is failing
```
Comment (slinkydeveloper) [inline kotlin/hello-world-lambda-cdk/lambda/src/main/resources/log4j2.properties:1]: Perhaps can you check whether we can configure this? https://github.com/restatedev/documentation/issues/184

File: kotlin/hello-world-lambda-cdk/lambda/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+Run:
+
+```shell
+./gradlew shadowJar
+```
+
+You'll find the shadowed jar in the `build` directory.
+
+The class to configure in Lambda is `dev.restate.sdk.examples.LambdaHandler`.
```
Comment (slinkydeveloper) [inline kotlin/hello-world-lambda-cdk/lambda/README.md:28]: Maybe in this readme say that the handler to configure is in lambda-jvm-cdk-stack.ts

File: kotlin/hello-world-lambda-cdk/lambda/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+* [Service interface definition `greeter.proto`](src/main/proto/greeter.proto)
+* [Service class implementation `Greeter`](src/main/kotlin/dev/restate/sdk/examples/Greeter.kt)
+* [Lambda handler `LambdaHandler`](src/main/kotlin/dev/restate/sdk/examples/LambdaHandler.kt)
+* [Test `GreeterTest`](src/test/kotlin/dev/restate/sdk/examples/GreeterTest.kt)
+* [Logging configuration](src/main/resources/log4j2.properties)
+
+## Download the example
+
+```shell
+wget https://github.com/restatedev/examples/releases/latest/download/kotlin-hello-world-lambda.zip && unzip kotlin-hello-world-lambda.zip -d kotlin-hello-world-lambda && rm kotlin-hello-world-lambda.zip
```
Comment (slinkydeveloper) [inline kotlin/hello-world-lambda-cdk/lambda/README.md:15]: Can you move this command in the upper directory readme?

File: kotlin/hello-world-lambda-cdk/lambda/README.md (hunk 2 above)
Comment (pcholakov) [reply kotlin/hello-world-lambda-cdk/lambda/README.md]: I'm sorry, I didn't realize I hit publish on the PR a bit too soon! I removed the file `lambda/README.md` as the top-level README should contain all the info.

File: kotlin/hello-world-lambda-cdk/lambda/README.md (hunk 3 above)
Comment (pcholakov) [reply kotlin/hello-world-lambda-cdk/lambda/README.md]: As above, sorry to waste your time :-(

File: kotlin/hello-world-lambda-cdk/lambda/src/main/resources/log4j2.properties (hunk 1 above)
Comment (pcholakov) [reply kotlin/hello-world-lambda-cdk/lambda/src/main/resources/log4j2.properties:1]: On it!

File: .github/workflows/pre-release.yml [hunk 4]
```diff
@@ -80,6 +80,13 @@ jobs:
           arguments: check
           build-root-directory: kotlin/hello-world-lambda
 
+      # CDK projects are a hybrid of TypeScript CDK stack + platform-specific handler code; the top-level npm CDK build
+      # is responsible for verifying the language-specific handler cod. They are also not a part of the TypeScript
+      # examples workspace, so we test them separately here.
+      - name: Test kotlin/hello-world-lambda-cdk
+        if: github.event.inputs.sdkTypescriptVersion != '' && github.event.inputs.sdkJavaVersion != ''
+        run: npm --prefix kotlin/hello-world-lambda-cdk run verify --workspaces
+
```
Comment (slinkydeveloper) [inline .github/workflows/pre-release.yml:89]: You need to copy paste this in the test.yml WF as well.

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -28,4 +28,5 @@ jobs:
             java-hello-world-http.zip
             java-hello-world-lambda.zip
             kotlin-hello-world-http.zip
-            kotlin-hello-world-lambda.zip
\ No newline at end of file
+            kotlin-hello-world-lambda.zip
+            kotlin-hello-world-lambda-cdk.zip
```
[the remark this answers is not in the document]
Comment (pcholakov) [reply .github/workflows/release.yml:32]: Totally missed that! Thank you!

**Review by slinkydeveloper — APPROVED** [review]

File: kotlin/hello-world-lambda-cdk/lambda/src/main/resources/log4j2.xml [hunk 6]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Configuration status="WARN">
+    <Appenders>
+        <Lambda name="Lambda" format="${env:AWS_LAMBDA_LOG_FORMAT:-TEXT}">
+            <LambdaTextFormat>
+                <PatternLayout>
+                    <pattern>%d{yyyy-MM-dd HH:mm:ss} %notEmpty{%X{AWSRequestId} }%notEmpty{[%X{restateServiceMethod}]}%notEmpty{[%X{restateInvocationId}]} %-5p %c - %m%n - customId=%X{customId}</pattern>
```
Comment (slinkydeveloper) [inline kotlin/hello-world-lambda-cdk/lambda/src/main/resources/log4j2.xml:7]: The customId i guess you don't need it anymore

Comment (pcholakov) [reply kotlin/hello-world-lambda-cdk/lambda/src/main/resources/log4j2.xml:7]: Doh, thanks for catching this!

**Pull request merged**